spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/reader/ValueRowReader.scala

```diff
@@ -14,7 +14,10 @@ class ValueRowReader[T: TypeConverter](columnRef: ColumnRef) extends RowReader[T
     * @param row row fetched from Cassandra
     * @param rowMetaData: column names available in the `row` */
   override def read(row: Row, rowMetaData: CassandraRowMetadata): T =
-    converter.convert(GettableData.get(row, columnRef.cqlValueName))
+    converter.convert(GettableData.get(
+      row,
+      columnRef.cqlValueName,
+      rowMetaData.codecs(columnRef.cqlValueName)))
```

Comment:
So this breaks for a few things, The following tests
[info] - should allow to transform rows into KV pairs of two single-column types *** FAILED *** (145 milliseconds)
[info] - should allow to use of keyByAndApplyPartitioner *** FAILED *** (419 milliseconds)

Because the ValueReader is also used by KeyValueReader
https://github.com/datastax/spark-cassandra-connector/blob/44de124012698ab6bb985e5c45af5eaba4146cac/spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/reader/KeyValueRowReader.scala#L13-L15